libcontainer/rootfs_linux.go

```diff
@@ -594,6 +595,37 @@ func createDevices(config *configs.Config) error {
 }
 
 func bindMountDeviceNode(dest string, node *configs.Device) error {
+	var st unix.Stat_t
+	if err := unix.Lstat(dest, &st); err != nil {
+		if !os.IsNotExist(err) {
+			return errors.Wrapf(err, "could not stat existing path for device node: %q", dest)
+		}
+	} else if os.FileMode(st.Mode)&os.ModeSymlink == os.ModeSymlink {
```

Comment:
My bad on this one as well. This is also wrong (it has the same misunderstanding). This should be `st.Mode&unix.S_IFMT == unix.S_IFLNK`. And you might argue that we should handle regular files here too, I don't know what the intended semantics are but if unlink symlinks maybe we should also unlink regular files? Personally I'd prefer if we didn't do this (assuming it doesn't break any containers).

Reply:
updated